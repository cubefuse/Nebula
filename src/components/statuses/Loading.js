// @flow
import React from "react";
import { Layout, Spin } from "antd";

const { Content } = Layout;

type Props = {
  message?: string
};

/**
 * Loading indicator full-page component.
 */
export default class Loading extends React.Component<Props> {
  render() {
    return (
      <Content style={{ textAlign: "center" }}>
        <Spin size="large" />
        <h1 style={{ fontSize: "xx-large", color: "gray" }}>
          {this.props.message}
        </h1>
      </Content>
    );
  }
}
