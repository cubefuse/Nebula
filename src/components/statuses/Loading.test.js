import React from "react";
import "jest-dom/extend-expect";
import { render } from "react-testing-library";
import Loading from "./Loading";

test("can render", () => {
  const node = render(<Loading />);
  expect(node).toMatchSnapshot();
});

test("can render with loading message", () => {
  const node = render(<Loading message="This is a test loading message" />);
  expect(node).toMatchSnapshot();
});
