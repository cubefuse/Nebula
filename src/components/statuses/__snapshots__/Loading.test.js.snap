// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render 1`] = `
Object {
  "container": <div>
    <div
      class="ant-layout-content"
      style="text-align: center;"
    >
      <div
        class="ant-spin ant-spin-lg ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i />
          <i />
          <i />
          <i />
        </span>
      </div>
      <h1
        style="font-size: xx-large; color: gray;"
      />
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`can render with loading message 1`] = `
Object {
  "container": <div>
    <div
      class="ant-layout-content"
      style="text-align: center;"
    >
      <div
        class="ant-spin ant-spin-lg ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i />
          <i />
          <i />
          <i />
        </span>
      </div>
      <h1
        style="font-size: xx-large; color: gray;"
      >
        This is a test loading message
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
